metafacture-io/src/test/java/org/metafacture/io/SruOpenerTest.java

```diff
@@ -0,0 +1,196 @@
+package org.metafacture.io;
+
+import org.junit.*;
+import org.metafacture.framework.MetafactureException;
+import org.metafacture.framework.ObjectReceiver;
+
+import com.github.tomakehurst.wiremock.client.MappingBuilder;
+import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import com.github.tomakehurst.wiremock.http.RequestMethod;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import com.github.tomakehurst.wiremock.matching.RequestPatternBuilder;
+import com.github.tomakehurst.wiremock.matching.StringValuePattern;
+import com.github.tomakehurst.wiremock.matching.UrlPattern;
+import org.metafacture.framework.helpers.DefaultObjectPipe;
+import org.metafacture.framework.helpers.DefaultObjectReceiver;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Arrays;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+
+public final class SruOpenerTest {
+
+    private static StringBuilder resultCollector = new StringBuilder();
+    private static final String RESPONSE_BODY = "response bödy"; // UTF-8
+    private static final String TEST_URL = "/test/path";
+    private static SruOpener sruOpener = new SruOpener();
+
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(WireMockConfiguration.wireMockConfig().
+                                                                             jettyAcceptors(Runtime.getRuntime()
+                                                                                                   .availableProcessors())
+                                                                             .dynamicPort());
+
+    @Mock
+    private ObjectReceiver<Reader> receiver;
+
+    public SruOpenerTest() {
+    }
+
+    @Before
+    public void setUp() {
+        sruOpener = new  SruOpener();
+        final char[] buffer = new char[ 1024 * 1024 * 16];
+        sruOpener.setReceiver(new DefaultObjectPipe<Reader, ObjectReceiver<String>>() {
+            @Override
+            public void process(final Reader reader) {
+                int size;
+                try {
+                    while ((size = reader.read(buffer)) != -1) {
+                        int offset = 0;
+                        for (int i = 0; i < size; ++i) {
+                            resultCollector.append(buffer, offset, size - offset);
+                            offset = i + 1;
+                        }
```

Comment:
What is this loop supposed to do? It seems to append _all_ suffixes of `buffer` to `resultCollector`, thus resulting in an OOM in `test()`.

```suggestion
                        resultCollector.append(buffer, 0, size);
```